File: ios/Cache/State/PinChangeOperationState.swift
//
// Copyright © 2023 Nevis Security AG. All rights reserved.
//

import NevisMobileAuthentication

class PinChangeOperationState: OperationState {
	// MARK: OperationState

	func cancel() {
		fatalError("Must override.")
	}
}

// MARK: -

class PinChange: PinChangeOperationState {
	// MARK: Properties

	let context: PinChangeContext?
	let handler: PinChangeHandler?

	// MARK: Initialization

	init(context: PinChangeContext?, handler: PinChangeHandler? = nil) {
		self.context = context
		self.handler = handler
	}

	// MARK: OperationState

	override func cancel() {
		handler?.cancel()
	}
}

File: android/src/main/java/ch/nevis/mobile/authentication/sdk/react/cache/state/PinChangeOperationState.kt
/*
 * Copyright © 2023 Nevis Security AG. All rights reserved.
 */

package ch.nevis.mobile.authentication.sdk.react.cache.state

import ch.nevis.mobile.sdk.api.operation.pin.PinChangeContext
import ch.nevis.mobile.sdk.api.operation.pin.PinChangeHandler

sealed class PinChangeOperationState : OperationState

data class PinChange(
    val pinChangeContext: PinChangeContext?,
    val pinChangeHandler: PinChangeHandler?,
) : PinChangeOperationState() {

    override fun cancel() {
        pinChangeHandler?.cancel()
    }
}
